app/controllers/events_controller.rb

```diff
@@ -0,0 +1,34 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @events = Event.all.order('created_at DESC')
+    @future_events = Event.where('start_time > ?', Time.now)
+    @previous_events = Event.where('start_time < ?', Time.now)
```

Comment:
Refactor the @future_events and @previous_events to use your event scopes:
- lists past events, i.e. `@previous_events = Event.previous`
-  lists ucomping events, i.e. `@future_events = Event.upcoming`